src/lex.cpp

```diff
@@ -722,16 +734,37 @@ const char8* lexer::end_of_previous_token() const noexcept {
 }
 
 void lexer::parse_hexadecimal_number() {
-  QLJS_ASSERT(this->is_hex_digit(this->input_[0]) || this->input_[0] == '.');
```

Comment:
This change seems unrelated to the rest of your changes. I filed issue #43 to fix this problem. Undo?